src/modules/secrets/Elsa.Secrets.Http/Services/OAuth2TokenService.cs

```diff
@@ -48,23 +48,25 @@ public async Task<TokenResponse> GetToken(Secret secret, string? authCode, strin
 			{ "grant_type", secret.GetProperty("GrantType") },
 			{ "client_id", clientId },
 			{ "client_secret", clientSecret },
-			{ "scope", secret.GetProperty("Scope") }
+			{ "offline_access ", "true" }
 		};
 
 		if (authCode != null)
 		{
 			content.Add("code", authCode);
 		}
+		else
+		{
+			content.Add("scope", secret.GetProperty("Scope"));
+		}
 		if (redirectUri != null)
 		{
 			content.Add("redirect_uri", redirectUri);
 		}
-
 		try
 		{
 			var httpClient = _httpClientFactory.CreateClient(nameof(OAuth2TokenService));
 			httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-			httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", Base64Encode(clientId + ":" + clientSecret));
```

Comment:
Note: OpenIddict supports sending the client credentials in the `Authorization` header (basic authentication) or in the request form, but as per the spec, it's illegal to send credentials using both methods (which is likely why you were getting an error).

That said, if this client is expected to be used with other vendors that don't support sending the credentials in the request form, this change might break them. You may want to add an option to control how the credentials are sent 😃

Reply:
Yep, I indeed got an error, because the credentials were specified in 2 ways. Do you think every there are vendors, that support only credentials in request form and now in headers?